### PR TITLE
Add unit test for finalizer

### PR DESCRIPTION
**Pull request by Parship999**
### Ⅰ. Describe what this PR does
- The test scenarios covered in our finalizer test suite:
  - Add finalizer operations - testing various scenarios when adding finalizers
  - Remove finalizer operations - Testing various scenarios when removing finalizers
  - Error Handling - testing error conditions and edge cases
  - Object type compatibility - testing with different kubernetes object types
  - Edge cases - testing boundary conditions and special cases
  - Internal behavior - testing the underlying implementation details
 
- Test Coverage:
  - 100% code coverage
  - all test cases pass successfully

### Ⅱ. Does this pull request fix one issue?
- Resolves part of #2074

### CODE REVIEW

**Comment by ghost** [comment]
@ABNER-1 @furykerry could you please review this when possible?

File: pkg/util/finalizer_test.go [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+	assert.NoError(t, err)
+
+	// Verify empty finalizer was added
+	updatedPod := &corev1.Pod{}
+	err = fakeClient.Get(context.TODO(), types.NamespacedName{
+		Name:      pod.Name,
+		Namespace: pod.Namespace,
+	}, updatedPod)
+	assert.NoError(t, err)
+	assert.Contains(t, updatedPod.Finalizers, "")
```
Comment (furykerry) [inline pkg/util/finalizer_test.go]: we should return error if empty finalizer name is given

**Review by furykerry — APPROVED** [review]
/lgtm
/approve
